src/controllers/bulk/bulkenumerator.cpp

```diff
@@ -21,14 +21,14 @@ BulkEnumerator::~BulkEnumerator() {
     libusb_exit(m_context);
 }
 
-static bool is_interesting(struct libusb_device_descriptor *desc) {
-    for (int i = 0; bulk_supported[i].vendor_id; ++i) {
-        if ((bulk_supported[i].vendor_id == desc->idVendor) &&
-            (bulk_supported[i].product_id == desc->idProduct)) {
-            return true;
-        }
-    }
-    return false;
+static bool is_interesting(const libusb_device_descriptor& desc) {
+    auto vendorId = desc.idVendor;
+    auto productId = desc.idProduct;
+    return std::any_of(std::cbegin(bulk_supported),
+            std::cend(bulk_supported),
+            [=](const auto& dev) {
+                return dev.vendor_id == vendorId && dev.product_id == productId;
+            });
```

Comment:
This is a bit verbose without giving infos. How about this: 
```suggestion
    return std::any_of(std::cbegin(bulk_supported),
            std::cend(bulk_supported),
            [&](const auto& dev) {
                return dev.vendor_id == desc.vendorId && 
                    dev.product_id == desc.productId;
            });
```